test/unit/math/rev/core/ad_stack_allocator_test.cpp

```diff
@@ -0,0 +1,25 @@
+#include <stan/math/rev/core.hpp>
+#include <gtest/gtest.h>
+#include <vector>
+
+void ad_stack_allocator_test() {
+  std::vector<int, stan::math::AD_stack_allocator<int>> v;
+  v.reserve(3);
+  for (int i = 0; i < 5; i++) {
+    v.push_back(1);
+  }
+
```

Comment:
Can you sprinkle around in appropriate places tests like:

```EXPECT_TRUE(stan::math::ChainableStack::instance_->memalloc_.in_stack(v.data()));```

Or use these: https://github.com/stan-dev/math/blob/develop/test/unit/math/rev/util.hpp

That'll just make sure we're allocating things in the right place.

Reply:
Done. I can't directly use the utils function, as it expect that scalar is var.